### DISCUSSION TITLE
Heat exchanger

### DISCUSSION THREAD

**Discussion by saturngreen67** [idaholab/moose#26701] [Q&A Modules: General]
Hello everyone,

For my master thesis I need to add a coaxial heat exchanger inside my geological model which was done using GOLEM. 
![REM_sonde_coaxiale](https://github.com/idaholab/moose/assets/119292777/04ac8040-f80f-472d-83fa-b5c27626f1d7)
I need to investigate heat storage in earth for that I need circulate water inside the tubes. But I am fairly new to MOOSE. Do you know whether I can do it inside MOOSE without using any modules or do I need a particular module to do so. Also I need the temperature of water upon entrance to the outer pipe where it is being injected to be constant. How can I do this? I appreciate your advice.

Best regards,

**Comment by GiudGiud** [comment]
@1runer @cpgr can you please @saturngreen67 to some porous flow + thermal hydraulics modules examples?

**Reply by 1runer** [reply]
Hi @saturngreen67 

this sounds like an interesting application and quite extensive for a standard master thesis.

GOLEM is a MOOSE-based application that mainly simplifies what is possible with MOOSE [PorousFlow module](https://mooseframework.inl.gov/modules/porous_flow/)

The difficulty for you is that in geothermal applications the borehole is mostly simplified including a 1D line (GOLEM) or PolyLineSink (PorousFlow). Therefore it is impossible to represent the internal structure of a coaxial well within.

You would have different options:
- Explicit meshing of the coaxial hole with 3D elements and various blocks that assign the material properties for the inner or outer loop. This results in a large number of elements and a fairly fine meshing.
- Keeping the GOLEM model and using the [MultiApp](https://mooseframework.inl.gov/syntax/MultiApps/index.html) system. Calculating the wellbore flow  and internal heat exchange in a sub-app like [Thermal hydraulics](https://mooseframework.inl.gov/modules/thermal_hydraulics/index.html) and connecting it just in the bottom-most point of your borehole with the reservoir (exchange of temperature and pressure). A comparable example connecting a fracture via MultiApp is explained [HERE](https://mooseframework.inl.gov/modules/porous_flow/multiapp_fracture_flow_introduction.html)

I don't know of any example connecting PorousFlow and Thermal hydraulics. 
@GiudGiud Probably something similar with Navier-Stokes and Thermal hydraulics?

Also have a look at this [Discussion](https://github.com/idaholab/moose/discussions/26697). Even not fully solved, he uses the MultiApp approach for something similar.

**Reply by saturngreen67** [reply]
Thank you very much for your prompt response. I will try your suggestions and will get back to you.

**Reply by saturngreen67** [reply]
@1runer Thank you for your last reply. I had a meeting with my supervisor and she told me that I should put a 2D heat exchanger inside my 3D GOLEM model. Do you have any ideas on how it can be implemented or where I should begin?

**Reply by GiudGiud** [reply]
You can use the same approach (multiapp) with a 2D heat exchanger.

**Reply by 1runer** [reply]
What is the advantage of a 2D heat exchanger in a 3D domain?
I'm not aware how you properly scale the variables for the missing 3rd dimension.
 3D in 3D could be an option, but you will create a large number of elements with high computational expense. 

I recommend using either MultiApp ( as @GiudGiud ) suggested or defining a conceptual model where you can represent it as 1D line.

@saturngreen67 keep us updated about the approach to solve your problem.

**Reply by GiudGiud** [reply]
I imagine the 2D shape is closer to the real shape of the exchanger (thin) than the 1D shape. 
Intensive variables (temperature, pressure) would not need to be scaled. @1runer is right that for extensive variables (mass flow, energy) you will need to make sure the approach is conservative

**Reply by saturngreen67** [reply]
@1runer @GiudGiud Thank you both, I will try my options and keep you informed of my final approach
